packages/gatsby-plugin-page-creator/src/get-match-path.ts

```diff
@@ -0,0 +1,19 @@
+interface IOptionalMatchPath {
+  matchPath?: string
+}
+
+// Does the following transformations:
+//   `/foo/[id]/` => `/foo/:id`
+//   `/foo/[...id]/` => `/foo/*id`
+//   `/foo/[...]/` => `/foo/*`
+export function getMatchPath(srcPagesPath: string): IOptionalMatchPath {
+  if (srcPagesPath.includes(`[`) === false) return {}
+
+  return {
+    matchPath: srcPagesPath
+      .replace(`[...`, `*`)
+      .replace(`[`, `:`)
+      .replace(`]`, ``)
```

Comment:
`replace` only replaces the very first occurence. So this URL:

```
/products/[:id]/[...page]/
```

Is transformed to:
```
 /products/:id/*page]
```

It probably requires a regex in this case.

Reply:
Good catch, I got a fix for this! Thanks